scene/2d/tile_map.cpp

```diff
@@ -40,175 +40,291 @@
 #include "servers/navigation_server_3d.h"
 #endif // DEBUG_ENABLED
 
-Vector2i TileMapLayer::_coords_to_quadrant_coords(const Vector2i &p_coords) const {
-	int quad_size = get_effective_quadrant_size();
+#define TILE_MAP_DEBUG_QUADRANT_SIZE 16
 
-	// Rounding down, instead of simply rounding towards zero (truncating).
+/////////////////////////////// Debug //////////////////////////////////////////
+Vector2i TileMapLayer::_coords_to_debug_quadrant_coords(const Vector2i &p_coords) const {
+	const int debug_quadrant_size = TILE_MAP_DEBUG_QUADRANT_SIZE;
 	return Vector2i(
-			p_coords.x > 0 ? p_coords.x / quad_size : (p_coords.x - (quad_size - 1)) / quad_size,
-			p_coords.y > 0 ? p_coords.y / quad_size : (p_coords.y - (quad_size - 1)) / quad_size);
+			p_coords.x > 0 ? p_coords.x / debug_quadrant_size : (p_coords.x - (debug_quadrant_size - 1)) / debug_quadrant_size,
+			p_coords.y > 0 ? p_coords.y / debug_quadrant_size : (p_coords.y - (debug_quadrant_size - 1)) / debug_quadrant_size);
 }
 
-HashMap<Vector2i, TileMapQuadrant>::Iterator TileMapLayer::_create_quadrant(const Vector2i &p_qk) {
-	TileMapQuadrant q;
-	q.coords = p_qk;
+void TileMapLayer::_debug_update() {
```

Comment:
If it's debug then it should be all wrapped in `#ifdef DEBUG_ENABLED` I think. Only parts of the debug update are conditional...

Reply:
I don't know. From what I could read, it seems like it is not done for drawing collision shape in general. So, I am not sure we can add a condition here.

Reply:
Collision shapes can only draw in debug build.
What I mean is that the debug code should not run at all in release builds. Meantime the `_debug_update()` is always called and wastes performance when not needed.

Reply:
> Collision shapes can only draw in debug build.

If that's the case, then yes, I can wrap the code in `#ifdef DEBUG_ENABLED` then. I'll do that.